### PR TITLE
Allow changing the font, font size, and font outline of hotkey text.

### PR DESCRIPTION
**Pull request by kalbert312**
Allows changing the font, size, and outline of the hotkey font per bar.

hotkey-font

TODO:
- [x] Use tagged version of libs

### CODE REVIEW

File: .pkgmeta [hunk 1]
```diff
@@ -10,13 +10,15 @@ externals:
...
   libs/AceGUI-3.0: https://repos.wowace.com/wow/ace3/trunk/AceGUI-3.0
+  libs/AceGUI-3.0-SharedMediaWidgets: https://repos.wowace.com/wow/ace-gui-3-0-shared-media-widgets
   libs/AceConfig-3.0: https://repos.wowace.com/wow/ace3/trunk/AceConfig-3.0
   libs/LibActionButton-1.0: https://repos.wowace.com/wow/libactionbutton-1-0
   libs/LibButtonGlow-1.0: https://repos.wowace.com/wow/libbuttonglow-1-0
   libs/LibKeyBound-1.0: https://repos.wowace.com/wow/libkeybound-1-0/trunk/LibKeyBound-1.0
   libs/LibDBIcon-1.0: https://repos.wowace.com/wow/libdbicon-1-0/trunk/LibDBIcon-1.0
   libs/LibWindow-1.1: https://repos.wowace.com/wow/libwindow-1-1/trunk/LibWindow-1.1
   libs/LibDualSpec-1.0: https://repos.wowace.com/wow/libdualspec-1-0
+  libs/LibSharedMedia-3.0: https://repos.wowace.com/wow/sharedmedia/trunk
```
Comment (kalbert312) [inline .pkgmeta:21]: TODO: - [ ] tagged version

File: ButtonBar.lua [hunk 2]
```diff
@@ -16,6 +16,9 @@ local defaults = Bartender4:Merge({
 	rows = 1,
 	hidemacrotext = false,
 	hidehotkey = false,
+	hotkeyfont = "Arial Narrow",
+	hotkeyfontsize = 15,
+	hotkeyfontoutline = "OUTLINE",
```
Comment (kalbert312) [inline ButtonBar.lua:21]: These are the defaults in `LibActionButton-1.0`.

**Comment by Nevcairiel** [comment]
The defaults are problematic because there is not one default font, but a class of default fonts. The default is "NumberFontNormalSmallGray", which can resolve to 4 different fonts based on the current locale (asian languages use a different font), so using Arial as a default for everyone would potentially be a step backwards for those users.

Additionally, I also feel like the font change might be better managed through LibActionButton, but that would be a small change to introduce the font change there. The default is a bigger concern right now.

**Comment by kalbert312** [comment]
I'm new to wow modding... all of these lib repos appear to be on wowace. Is there mirroring between github and wowace repos such that https://github.com/ls-/LibActionButton-1.0 is a source of truth for the lib and I could just create a PR? I actually did give this a thought but the potential technical hurdle here and lack of ecosystem knowledge here demotivated me from pursuing.

**Comment by Nevcairiel** [comment]
LibActionButton-1.0 is not on GitHub, it has its Git repository on wowace. I've been meaning to also pull it onto github for a while..

**Comment by kalbert312** [comment]
Hmm I noticed an issue... not sure what causes it yet but if I change the font then assign a hotkey to button with out an action on it (spell, macro), then move an action to it, the hotkey font is the old one. Doesn't seem to happen 100%. Is it possible that `self.buttons` is not actually all the buttons?

**Comment by Nevcairiel** [comment]
An alternate solution that integrates with LibActionButton has been implemented now.

**Pull request closed without merging**